src/main/java/com/revealprecision/revealserver/service/MetaDataJdbcService.java

```diff
@@ -0,0 +1,142 @@
+package com.revealprecision.revealserver.service;
+
+import com.revealprecision.revealserver.enums.EntityStatus;
+import com.revealprecision.revealserver.props.BusinessStatusProperties;
+import com.revealprecision.revealserver.util.UserUtils;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MetaDataJdbcService {
+
+  @Autowired
+  JdbcTemplate jdbcTemplate;
+
+  @Autowired
+  BusinessStatusProperties businessStatusProperties;
+
+  @Autowired
+  EntityTagService entityTagService;
+
+
+  public Map<String, Pair<Class<?>, Object>> getMetadataFor(String entityTable, UUID identifier) {
+
+    String metadataTable = entityTable + "_metadata";
+
+    String query = "SELECT t.key,t.value->>'value',t.value->>'entity_tag',et.value_type From (\n"
+        + "SELECT  (jsonb_each(lm.entity_value)).*  from " + entityTable + " l\n" + "inner join "
+        + metadataTable+" lm on l.identifier = lm." + entityTable + "_identifier\n"
+        + "where l.identifier = '" + identifier + "'\n"
+        + "\t) as t left join entity_tag et on et.identifier = (t.value->>'entity_tag')::uuid";
+
+    return jdbcTemplate.query(query, (rs, rowNum) -> getMapFromResultSet(rs)).stream()
+        .filter(Objects::nonNull).flatMap(mapOfKeyAndPair -> mapOfKeyAndPair.entrySet().stream())
+        .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+
+  }
+
+  public void createOrUpdateMetadata(UUID entityIdentifier, String entity, String tagName,
+      Object value, Class<?> entityDataType) {
+    String quotes = (entityDataType.equals(String.class) ? "'" : "");
+
+
+// upsert String resolves to below insert statement
```

Comment:
are this comments or commented out code?

Reply:
Wanted to give an example of what the generated SQL will look like....